src/commands/publish/upload_form/file.rs

```diff
@@ -0,0 +1,5 @@
+#[derive(Debug)]
+pub struct File {
```

Comment:
we should evaluate what we're doing here! we should probably use a rust builtin... but the "name" here seems like it needs sorting

Reply:
got this out, getting what we need from the paths for all the things.